### PR TITLE
account_screen / fix display data when call api info user

### PR DESCRIPTION
**Pull request by hungvv-1301**
## Related Tickets
- [#Task #54665: Create View Layout Account](https://edu-redmine.sun-asterisk.vn/issues/54665)
- [#Task #54666: Create Logic Layout Account](https://edu-redmine.sun-asterisk.vn/issues/54666)
- [#Task #54667: Display profile ](https://edu-redmine.sun-asterisk.vn/issues/54667)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/drawable/bg_header_account_fragment.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <corners
+        android:bottomLeftRadius="@dimen/border_bottom_bg_header_account"
+        android:bottomRightRadius="@dimen/border_bottom_bg_header_account" />
+        <solid android:color="@color/main"/>
+        <stroke android:color="@color/main_700" android:width="@dimen/_1dp"/>
+</shape>
```
Comment (ghost) [inline app/src/main/res/drawable/bg_header_account_fragment.xml]: end line nhá

File: app/src/main/res/drawable/ic_help_24dp.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M12 6c-1.04 0-2.04 0.42-2.78 1.15C8.48 7.9 8.07 8.9 8.07 9.93h2C10.07 8.87 10.93 8 12 8c1.07 0 1.93 0.87 1.93 1.93 0 0.6-0.48 1.04-1.21 1.63-0.24 0.19-0.47 0.39-0.7 0.6-1 1-1.02 2.05-1.02 2.17V15h2v-0.63c0-0.02 0.03-0.39 0.44-0.8 0.15-0.15 0.34-0.3 0.54-0.45 0.77-0.63 1.95-1.59 1.95-3.19 0-1.04-0.41-2.04-1.15-2.78C14.04 6.42 13.04 6 12 6Zm-1 10h2v2h-2v-2Z"/>
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M12 2C6.49 2 2 6.49 2 12s4.49 10 10 10 10-4.49 10-10S17.51 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Z"/>
+</vector>
```
Comment (ghost) [inline app/src/main/res/drawable/ic_help_24dp.xml]: .

File: app/src/main/res/drawable/ic_ring_24dp.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M21.53 14.47l-2.03-2.03V9.75c0-1.86-0.7-3.65-1.94-5.03-1.25-1.38-2.96-2.24-4.81-2.43V0.75h-1.5v1.54C9.4 2.48 7.69 3.34 6.45 4.72 5.18 6.1 4.5 7.9 4.5 9.75v2.69l-2.03 2.03C2.33 14.61 2.25 14.8 2.25 15v2.25c0 0.2 0.08 0.39 0.22 0.53S2.8 18 3 18h5.25v0.58c-0.02 0.95 0.32 1.88 0.94 2.6 0.62 0.72 1.5 1.18 2.44 1.3 0.52 0.05 1.04 0 1.54-0.17 0.5-0.16 0.96-0.43 1.34-0.78 0.4-0.35 0.7-0.78 0.92-1.26 0.2-0.48 0.32-1 0.32-1.52V18H21c0.2 0 0.39-0.08 0.53-0.22s0.22-0.33 0.22-0.53V15c0-0.2-0.08-0.39-0.22-0.53Zm-7.28 4.28c0 0.6-0.24 1.17-0.66 1.6C13.17 20.75 12.6 21 12 21c-0.6 0-1.17-0.24-1.6-0.66-0.41-0.42-0.65-1-0.65-1.59V18h4.5v0.75Zm6-2.25H3.75v-1.19l2.03-2.03C5.92 13.14 6 12.95 6 12.75v-3c0-1.6 0.63-3.12 1.76-4.24C8.88 4.38 10.4 3.75 12 3.75c1.6 0 3.12 0.63 4.24 1.76C17.37 6.63 18 8.16 18 9.75v3c0 0.2 0.08 0.39 0.22 0.53l2.03 2.03v1.19Z"/>
+</vector>
```
Comment (ghost) [inline app/src/main/res/drawable/ic_ring_24dp.xml]: .

File: app/src/main/res/drawable/ic_statistical_24dp.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M20.25 12.14h-8.4V3.75c0-0.1-0.07-0.19-0.18-0.19h-0.6c-1.24 0-2.46 0.24-3.6 0.71C6.34 4.74 5.3 5.44 4.43 6.31c-0.85 0.85-1.53 1.86-2 2.98-0.5 1.15-0.74 2.4-0.74 3.65 0 1.23 0.24 2.45 0.7 3.59 0.48 1.13 1.17 2.17 2.04 3.04 0.86 0.86 1.87 1.53 2.98 2 1.16 0.5 2.4 0.74 3.65 0.74 1.23 0 2.45-0.24 3.6-0.7 1.13-0.48 2.16-1.17 3.03-2.04 0.86-0.86 1.54-1.87 2-2.98 0.5-1.16 0.75-2.4 0.75-3.65v-0.61c0-0.1-0.09-0.19-0.19-0.19Zm-3.71 6.32c-0.73 0.72-1.59 1.3-2.53 1.68-0.95 0.39-1.96 0.58-2.98 0.58-2.07-0.01-4.01-0.82-5.47-2.28-1.47-1.47-2.28-3.42-2.28-5.5S4.1 8.9 5.56 7.44c1.28-1.29 2.93-2.07 4.7-2.24v8.53h8.54c-0.18 1.8-0.96 3.45-2.26 4.73Zm5.77-7.62l-0.06-0.66c-0.2-2.16-1.16-4.2-2.7-5.74-1.54-1.54-3.57-2.49-5.75-2.7L13.14 1.7c-0.1-0.01-0.2 0.07-0.2 0.18v9c0 0.1 0.08 0.2 0.19 0.2l9-0.03c0.1 0 0.2-0.1 0.18-0.2Zm-7.78-1.37V3.46c1.47 0.3 2.83 1.04 3.9 2.1 1.07 1.07 1.8 2.43 2.1 3.9l-6 0.01Z"/>
+</vector>
```
Comment (ghost) [inline app/src/main/res/drawable/ic_statistical_24dp.xml]: .

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -15,7 +15,10 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.BookShop"
         tools:targetApi="31">
-
+        <activity
+            android:name=".ui.authentication.AuthenticationActivity"
+            android:exported="false"
+            android:noHistory="true" />
```
Comment (daolq-2712) [inline app/src/main/AndroidManifest.xml]: App này support xoay màn hình à em?

File: app/src/main/java/com/atom/android/bookshop/ui/account/AccountPresenter.kt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        })
+    }
+
+    override fun destroyToken(context: Context?) {
+        SharedPreferenceUtils.getInstance(context)?.destroyToken()
+    }
+
+
+    override fun onStart() {
+        // late impl
```
Comment (daolq-2712) [inline app/src/main/java/com/atom/android/bookshop/ui/account/AccountPresenter.kt]:
```suggestion
        // TODO implement later
```
Sửa hết cho mấy chỗ khác em nhé!

vì sao cmt như này em đọc ở đây: https://www.jetbrains.com/help/idea/using-todo.html

File: app/src/main/res/drawable/bg_header_account_fragment.xml (hunk 1 above)
Comment (daolq-2712) [inline app/src/main/res/drawable/bg_header_account_fragment.xml]:
```suggestion
</shape>

```

File: app/src/main/res/drawable/ic_help_24dp.xml (hunk 2 above)
Comment (daolq-2712) [inline app/src/main/res/drawable/ic_help_24dp.xml]:
```suggestion
</vector>

```

File: app/src/main/res/drawable/ic_ring_24dp.xml (hunk 3 above)
Comment (daolq-2712) [inline app/src/main/res/drawable/ic_ring_24dp.xml]:
```suggestion
</vector>

```

File: app/src/main/res/drawable/ic_statistical_24dp.xml (hunk 4 above)
Comment (daolq-2712) [inline app/src/main/res/drawable/ic_statistical_24dp.xml]:
```suggestion
</vector>

```

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

File: app/src/main/AndroidManifest.xml (hunk 5 above)
Comment (hungvv-1301) [reply app/src/main/AndroidManifest.xml]: em set portrain lại ạ

File: app/src/main/java/com/atom/android/bookshop/ui/main/MainActivity.kt [hunk 7]
```diff
@@ -1,46 +1,110 @@
...
+        val intent = Intent(this@MainActivity, AuthenticationActivity::class.java)
+        startActivity(intent)
+        finish()
+    }
+
+    companion object {
+        const val POSITION_FRAGMENT_HOME = 0
+        const val POSITION_FRAGMENT_DISCOUNT = 1
+        const val POSITION_FRAGMENT_BILL = 2
+        const val POSITION_FRAGMENT_ACCOUNT = 3
```
Comment (vietnh-1109) [inline app/src/main/java/com/atom/android/bookshop/ui/main/MainActivity.kt:108]: ko dùng đâu nữa thì để private đi e

File: app/src/main/res/layout/fragment_account.xml [hunk 8]
```diff
@@ -3,11 +3,209 @@
...
+            android:text="@string/text_logout"
+            android:textAllCaps="true"
+            android:textColor="@color/white"
+            android:textStyle="bold"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+
+
```
Comment (vietnh-1109) [inline app/src/main/res/layout/fragment_account.xml]: remove

File: app/src/main/java/com/atom/android/bookshop/ui/main/MainActivity.kt (hunk 7 above)
Comment (hungvv-1301) [reply app/src/main/java/com/atom/android/bookshop/ui/main/MainActivity.kt:108]: Dạ có dùng ở ngoài class ạ

**Pull request merged**